test/test-plan/eventing/content-based-routing.md

```diff
@@ -0,0 +1,143 @@
+> > distinguishing between these deliveries. The implementation MUST NOT modify
+> > the event data in this process.
+
+> > Reply events generated during event delivery MUST be re-enqueued by the
+> > Broker using the same routing and persistence as events delivered to the
+> > Broker's Addressable URL. Reply events re-enqueued in this manner MUST be
+> > evaluated against all Triggers associated with the Broker, including the
+> > Trigger that generated the reply. If the storage of the reply event in the
+> > Broker fails, the entire event delivery MUST be failed and the delivery to
+> > the Trigger's subscriber MUST be retried. Implementations MAY implement
+> > event-loop detection; it is RECOMMENDED that any such controls be documented
+> > to end-users. Implementations MAY avoid using HTTP to deliver event replies
+> > to the Broker's event-delivery input and instead use an internal queueing
+> > mechanism.
+
+# Testing Content Based Routing Conformance:
+
+We are going to be testing that, under normal operation (no pod failures), when
+an event is sent to a broker, each associated Trigger receives the event exactly
+once when it passes the trigger filter. We are going to test different filters.
+
+We are going to send events to the event-display service to manually check that
+events are being received.
+
+You can find the resources for running these tests inside the
+[control-plane/content-based-routing](control-plane/content-based-routing)
+directory.
+
+## [Pre] Creating a Broker
+
+```sh
+kubectl apply -f control-plane/content-based-routing/trigger.yaml
+```
+
+Expected output:
+
+```sh
+broker.eventing.knative.dev/conformance-broker created
+```
+
+## [Pre] Creating an Event Display service
+
+```sh
+kubectl apply -f control-plane/content-based-routing/event-display.yaml
+```
+
+Expected output:
+
+```sh
+deployment.apps/event-display created
+service/event-display created
+```
+
+Check that the Event Display deployment is ready:
+
+```sh
+kubectl get deployment event-display
+NAME            READY   UP-TO-DATE   AVAILABLE   AGE
+event-display   1/1     1            1           3s
+```
+
+## [Pre] Creating a Trigger with no filters
+
+```sh
+kubectl apply -f control-plane/content-based-routing/trigger-no-filter.yaml
+```
+
+Expected output:
+
+```sh
+trigger.eventing.knative.dev/conformance-trigger-no-filter created
+```
+
+Check the Trigger is ready:
+
+```sh
+kubectl get triggers.eventing.knative.dev conformance-trigger-no-filter -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Expected output:
+
+```sh
+True
+```
+
+## [Test] Send an event to the Broker
+
+Sends an event using `kn event`
+
+```sh
+kn event send --type dev.knative.conformance.ping --id 1 --to Broker:eventing.knative.dev/v1:conformance-broker
+```
+
+Check the event has been receive once by the Event Display service:
+
+```sh
+kubectl logs -l=app=event-display --tail=-1
+```
+
+Expected output:
```

Comment:
@lionelvillard same here: 

```
{
  "test": "control-plane/content-based-routing/event-delivery-to-broker"
  "output": {
  	"receivedCloudEvent": {
            <JSON structure of the CloudEvent received here>
        },
  }
}
```

By having this output we can create a small program that compares inputs and outputs to later validate conformance automatically.